### ISSUE TITLE
6.x configure button colors not changed the color

### ISSUE THREAD

**Issue by jminutaglio** [ivanvorobei/SPPermissions#233]
Not sure if this is a bug or a 5.x -> 6.x migration issue on my part, but:

in:
`func configure`

in 5.x, the following would turn the icons and buttons purple:
```
// Colors
        cell.iconView.color = .systemPurple
        cell.button.allowedBackgroundColor = .systemPurple
        cell.button.allowTitleColor = .systemPurple
```

In 6x, the following turns the icons purple, but not the buttons:
```
// Colors
        cell.permissionDrawIconView.tintColor = .systemPurple
        cell.permissionButton.allowedBackgroundColor = .systemPurple
        cell.permissionButton.allowTitleColor = .systemPurple
```
        
Is that syntax incorrect or is it a bug w/ the color configuration?  Thank you in advance!

**Comment by jminutaglio** [comment]
More info:

Button is initially blue in allowed state, but if you tap it it turns purple.

This is different than the 5.x behavior.

Hope this helps.

"initial view":
<img width="341" alt="Screen Shot 2021-05-19 at 12 07 27 PM" src="https://user-images.githubusercontent.com/49879291/118846634-ed430980-b89a-11eb-8233-187fa249e9c4.png">

"after tap":
<img width="341" alt="Screen Shot 2021-05-19 at 12 07 36 PM" src="https://user-images.githubusercontent.com/49879291/118846650-f16f2700-b89a-11eb-8048-9ba820e477b8.png">

**Comment by ivanvorobei** [comment]
Thanks, true. I should add update interface after custom configuration. 

I have added it already, so you can update to `6.0.9` and it will work. 

Thanks for your reports!

**Comment by jminutaglio** [comment]
I think 6.0.9 introduced a new bug:

the conformance to delegate is:
<img width="721" alt="Screen Shot 2021-05-19 at 1 29 22 PM" src="https://user-images.githubusercontent.com/49879291/118857736-5ed48500-b8a6-11eb-9c16-53d1feb2aeb4.png">

Which doesn't allow for a cell `SPPermissionsTableViewCell` to be returned...

vs:
<img width="721" alt="Screen Shot 2021-05-19 at 1 32 44 PM" src="https://user-images.githubusercontent.com/49879291/118858079-bf63c200-b8a6-11eb-9903-2f8b6ecd2925.png">

**Comment by ivanvorobei** [comment]
Its true, no need return cell. You can change here simple, without return. 
I updated readme too.

**Comment by jminutaglio** [comment]
ok thx.

I'll work to figure this out but if you have any quick ideas let me know pls.

this:
```
// Colors
        cell.permissionDrawIconView.tintColor = .systemPurple
        cell.permissionButton.allowedBackgroundColor = .systemPurple
        cell.permissionButton.allowTitleColor = .systemPurple
```

results in black/blank (or transparent) buttons, and the icon tint which was working in 6.0.8, is not working now:
<img width="352" alt="Screen Shot 2021-05-19 at 1 44 09 PM" src="https://user-images.githubusercontent.com/49879291/118859656-7a408f80-b8a8-11eb-8bbf-d879a5fe068e.png">

**Comment by ivanvorobei** [comment]
For me is good
Can you please create empty project and attach here?
![Simulator Screen Shot - iPhone 12 Pro - 2021-05-19 at 21 00 49](https://user-images.githubusercontent.com/10995774/118861619-5f404080-b8e5-11eb-84ed-fea72cfb1730.png)

**Comment by jminutaglio** [comment]
I'm seeing that my `func configure ` was being called in 6.0.8, but isn't in 6.0.9...

Let me debug further on my side...

**Comment by ivanvorobei** [comment]
See, protocol changed. Maybe you left old version? Let me know if need any help.

**Comment by jminutaglio** [comment]
I could use some help...  :/

6.0.8 configure:
```
func configure(_ cell: SPPermissionsTableViewCell, for permission: SPPermissions.Permission) -> SPPermissionsTableViewCell {
        
        cell.permissionButton.allowTitle = "Allow"
        cell.permissionButton.allowedTitle = "Allowed"

        // Colors
        cell.permissionDrawIconView.tintColor = .systemPurple
        cell.permissionButton.allowedBackgroundColor = .systemPurple
        cell.permissionButton.allowTitleColor = .systemPurple
        
        // If you want set custom image.
        //cell.set(UIImage(named: "IMAGE-NAME")!)
        switch permission.type {
        case .camera:
            cell.permissionTitleLabel.text = "Camera"
            cell.permissionDescriptionLabel.text = "for photos & selfies"
            break
        case .photoLibrary:
            cell.permissionTitleLabel.text = "Photo Library"
            cell.permissionDescriptionLabel.text = "to add photos"
            break
        case .notification:
            cell.permissionTitleLabel.text = "Notification"
            cell.permissionDescriptionLabel.text = "to get notifications from App"
            break
        case .microphone:
            break
        case .calendar:
            break
        case .contacts:
            break
        case .reminders:
            break
        case .speech:
            break
        case .locationWhenInUse, .locationAlways:
            cell.permissionTitleLabel.text = "Location"
            cell.permissionDescriptionLabel.text = "to find others near you"
            break
        case .motion:
            break
        case .mediaLibrary:
            cell.permissionTitleLabel.text = "Media Library"
            cell.permissionDescriptionLabel.text = "for playing songs"
            break
        default:
            break
        }
        
        return cell
    }
```

6.0.9 configure:
```
func configure(_ cell: SPPermissionsTableViewCell, for permission: SPPermissions.Permission) {
        
        cell.permissionButton.allowTitle = "Allow"
        cell.permissionButton.allowedTitle = "Allowed"

        // Colors
        cell.permissionDrawIconView.tintColor = .systemPurple
        cell.permissionButton.allowedBackgroundColor = .systemPurple
        cell.permissionButton.allowTitleColor = .systemPurple
        
        // If you want set custom image.
        //cell.set(UIImage(named: "IMAGE-NAME")!)
        switch permission.type {
        case .camera:
            cell.permissionTitleLabel.text = "Camera"
            cell.permissionDescriptionLabel.text = "for photos & selfies"
            break
        case .photoLibrary:
            cell.permissionTitleLabel.text = "Photo Library"
            cell.permissionDescriptionLabel.text = "to add photos"
            break
        case .notification:
            cell.permissionTitleLabel.text = "Notification"
            cell.permissionDescriptionLabel.text = "to get notifications from App"
            break
        case .microphone:
            break
        case .calendar:
            break
        case .contacts:
            break
        case .reminders:
            break
        case .speech:
            break
        case .locationWhenInUse, .locationAlways:
            cell.permissionTitleLabel.text = "Location"
            cell.permissionDescriptionLabel.text = "to find others near you"
            break
        case .motion:
            break
        case .mediaLibrary:
            cell.permissionTitleLabel.text = "Media Library"
            cell.permissionDescriptionLabel.text = "for playing songs"
            break
        default:
            break
        }
        
        return
    }
```

The 6.0.8 does what is expected.  The 6.0.9 doesn't seem to called (ie: no color changes or text changes).  Does anything look obvious?  

Thank you in advance.

**Comment by ivanvorobei** [comment]
Can you add print something inside, it calling or not?

If it called, confirm 2 things:
1) installed `6.0.9`
2) Set data source

**Comment by jminutaglio** [comment]
yes: `6.0.9`

Data source is set:
 ```
// Show dialog
                    DispatchQueue.main.async {
                        let controller = SPPermissions.dialog([.camera, .photoLibrary, .locationWhenInUse, .mediaLibrary, .notification])

                        
                        // Ovveride texts in controller
                        controller.titleText = "PERMISSIONS"
                        controller.headerText = "App requests these"
                        controller.footerText = "App needs these permissions for full functionality, but allowing them is your choice! (Push notifications are optional)"
                        
                        
                        // Set `DataSource` or `Delegate` if need.
                        // By default using project texts and icons.
                        controller.dataSource = self
                        controller.delegate = self

                        // Always use this method for present
                        controller.present(on: self)
                    }
```

I will add a print and verify...  more to come.

**Comment by jminutaglio** [comment]
on `6.0.10`

```
    func configure(_ cell: SPPermissionsTableViewCell, for permission: SPPermissions.Permission) {

        print("sppermissionsdatasournce - in the func!")
```

Is never printed.  IE: the `func configure` is not called.

**Comment by ivanvorobei** [comment]
Can you create empty project with minimum code please and send here?

**Comment by ivanvorobei** [comment]
Thanks for your help, its fixed in `6.1.0`. Hope it last my bug.

**Comment by jminutaglio** [comment]
ok great - TY so much!  :)
